core/src/main/java/org/infinispan/cache/impl/SimpleLocalCacheImpl.java

```diff
@@ -0,0 +1,1608 @@
+		Metadata.Builder builder = metadata.builder();
+		return builder != null ? builder.merge(defaultMetadata).build() : metadata;
+	}
+
+	private Metadata createMetadata(long lifespan, TimeUnit unit) {
+		return new EmbeddedMetadata.Builder().lifespan(lifespan, unit).maxIdle(configuration.expiration().maxIdle()).build();
+	}
+
+	private Metadata createMetadata(long lifespan, TimeUnit lifespanUnit, long maxIdleTime, TimeUnit maxIdleTimeUnit) {
+		return new EmbeddedMetadata.Builder()
+				.lifespan(lifespan, lifespanUnit)
+				.maxIdle(maxIdleTime, maxIdleTimeUnit)
+				.build();
+	}
+
+	@Override
+	public AdvancedCache<K, V> withFlags(Flag... flags) {
+		// the flags are mostly ignored
+		return this;
+	}
+
+	@Override
+	public void addInterceptor(CommandInterceptor i, int position) {
+		throw log.interceptorStackNotSupported();
+	}
+
+	@Override
+	public boolean addInterceptorAfter(CommandInterceptor i, Class<? extends CommandInterceptor> afterInterceptor) {
+		throw log.interceptorStackNotSupported();
+	}
+
+	@Override
+	public boolean addInterceptorBefore(CommandInterceptor i, Class<? extends CommandInterceptor> beforeInterceptor) {
+		throw log.interceptorStackNotSupported();
+	}
+
+	@Override
+	public void removeInterceptor(int position) {
+		throw log.interceptorStackNotSupported();
+	}
+
+	@Override
+	public void removeInterceptor(Class<? extends CommandInterceptor> interceptorType) {
+		throw log.interceptorStackNotSupported();
+	}
+
+	@Override
+	public List<CommandInterceptor> getInterceptorChain() {
+		return Collections.EMPTY_LIST;
+	}
+
+	@Override
+	public EvictionManager getEvictionManager() {
+		return getComponentRegistry().getComponent(EvictionManager.class);
+	}
+
+	@Override
+	public ExpirationManager<K, V> getExpirationManager() {
+		return getComponentRegistry().getComponent(ExpirationManager.class);
+	}
+
+	@Override
+	public ComponentRegistry getComponentRegistry() {
+		return componentRegistry;
+	}
+
+	@Override
+	public DistributionManager getDistributionManager() {
+		return getComponentRegistry().getComponent(DistributionManager.class);
+	}
+
+	@Override
+	public AuthorizationManager getAuthorizationManager() {
+		return getComponentRegistry().getComponent(AuthorizationManager.class);
+	}
+
+	@Override
+	public boolean lock(K... keys) {
+		throw log.lockOperationsNotSupported();
+	}
+
+	@Override
+	public boolean lock(Collection<? extends K> keys) {
+		throw log.lockOperationsNotSupported();
+	}
+
+	@Override
+	public void applyDelta(K deltaAwareValueKey, Delta delta, Object... locksToAcquire) {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public RpcManager getRpcManager() {
+		return null;
+	}
+
+	@Override
+	public BatchContainer getBatchContainer() {
+		return null;
+	}
+
+	@Override
+	public InvocationContextContainer getInvocationContextContainer() {
+		return null;
+	}
+
+	@Override
+	public DataContainer<K, V> getDataContainer() {
+		if (dataContainer == null) {
```

Comment:
It seems we should store the data container in a local variable before doing the null check this way a concurrent stop wouldn't give a NPE back possibly.